csharp/src/Ice/CancelableTask.cs

```diff
@@ -0,0 +1,52 @@
+//
+// Copyright (c) ZeroC, Inc. All rights reserved.
+//
+
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace ZeroC.Ice
+{
+    internal static class CancelableTask
```

Comment:
Added:
```
    /// <summary>CancelableTask provides WhenAny static methods which creates a task that is completed either when the
    /// given task completes or when the given cancellation token is canceled. This is useful when canceling the given
    /// task isn't an option. For example, the user might want to cancel an invocation that is waiting for connection
    /// establishment. Instead of canceling the connection establishment which might be shared by other invocations we
    /// just cancel the wait on the connection esablishement for the invocation. The same applies for invocations which
    /// are waiting on a connection to be sent.</summary>
```